### PR TITLE
Add native support for new (Slackware 15.0) local packages directory

### PR DESCRIPTION
**Pull request by aclemons**
This mimics the fallback behaviour for choosing a GPG version, by first
checking if the new packages location exists, otherwise checking for the
old packages directory.

### CODE REVIEW

File: slackroll
Comment (rg3) [inline slackroll]: The first os.path.sep argument here is not needed, I think, because you're already getting a full path from the yielding function. Other than that, everything looks good to me. If you've tested this under -current and 14.2 and you've seen the program choose both paths, that's perfect. :+1:

File: slackroll
Comment (rg3) [inline slackroll:550]: This change needs to be done for the comment at get_pkg_filelists too.

**Review by rg3 — CHANGES_REQUESTED** [review]
I've searched for "/var/log/packages" in other source files and there are quite a few mentions inside helpfiles:

* helpfiles/local-search.txt
* helpfiles/local-info.txt
* helpfiles/all.txt
* helpfiles/manifest-search.txt
* helpfiles/info.txt
* helpfiles/missing-search.txt

I think there are two reasonable options. The first one is to keep the documentation as it is, using the traditional and known path as long as the compatibility symlinks don't go away. This probably won't confuse the reader. When the symlinks are removed, most users should know about the new path and it should be updated.

The second option is coming up with a name for that directory and its contents, like "pkgtool's installed packages database" or a similar expression and use it consistently everywhere in the documentation.

Or if you have another better idea, it's your call.

**Comment by rg3** [comment]
@aclemons Ping! This PR was left with requested changes that were never completed. Are you still using Slackware?

**Comment by aclemons** [comment]
Yes I am. I'll update this PR and ping you back.

**Comment by rg3** [comment]
That's great to hear. Thanks, Andrew!

File: slackroll
Comment (aclemons) [reply slackroll:550]: I've updated this one

File: slackroll
Comment (aclemons) [reply slackroll]: I've removed this first one.

**Comment by aclemons** [comment]
Regarding the references to /var/log/packages in the help files, I started changing this to your suggestion but some of them make less sense then, i.e. local-info.txt

```
It was added because it is usually faster to type "slackroll local-info PKG"
than to run "less /var/log/packages/PKG-something", but other than that they
are mostly equivalent.
```
Changing that one won't help much.

I think the best would be to leave the help as is for now, let 15.0 be released (it has still not been released) and then potentially change it to the new directory, or even possibly wait for another release after that. I highly doubt Pat will take the backwards compatibility symlink out any time soon.

**Comment by rg3** [comment]
> Regarding the references to /var/log/packages in the help files, I started changing this to your suggestion but some of them make less sense then, i.e. local-info.txt
> 
> ```
> It was added because it is usually faster to type "slackroll local-info PKG"
> than to run "less /var/log/packages/PKG-something", but other than that they
> are mostly equivalent.
> ```
> 
> Changing that one won't help much.
> 
> I think the best would be to leave the help as is for now, let 15.0 be released (it has still not been released) and then potentially change it to the new directory, or even possibly wait for another release after that. I highly doubt Pat will take the backwards compatibility symlink out any time soon.

In a second look, that sounds reasonable, so lets go with that plan.

**Pull request merged**
